nibabel/casting.py

```diff
@@ -23,6 +23,15 @@ class CastingError(Exception):
 _test_val = 2**63 + 2**11  # Should be exactly representable in float64
 TRUNC_UINT64 = np.float64(_test_val).astype(np.uint64) != _test_val
 
+# np.sctypes is deprecated in numpy 2.0 and np.core.sctypes should not be used instead.
+sctypes = {
+    "int": [np.int8, np.int16, np.int32, np.int64],
+    "uint": [np.uint8, np.uint16, np.uint32, np.uint64],
+    "float": [np.float16, np.float32, np.float64, np.float128],
+    "complex": [np.complex64, np.complex128, np.complex256],
+    "others": [bool, object, bytes, str, np.void],
+}
+
```

Comment:
I'm not entirely convince this is enough to keep a working status on all platforms. It looks to me like `np.sctypes` is defined per platform, and if this is correct, we might need to add some logic to define the mapping between strings and scalar types per platform as well.